exe/wallet/http-bridge/Main.hs

```diff
@@ -106,144 +122,186 @@ import qualified Network.Wai.Handler.Warp as Warp
                               Main entry point
 -------------------------------------------------------------------------------}
 
-main :: IO ()
-main = runCli withBackend cliDefinition
-
-withBackend :: Manager -> Environment -> IO ()
-withBackend manager env@Environment {..} = do
-    listen <- parseWalletListen env
-    backendPort <- parseArg $ longOption "backend-port"
-    dbFile <- parseOptionalArg $ longOption "database"
-    stateDir <- parseOptionalArg $ longOption "state-dir"
-    let verbosity = verbosityFromArgs args
-    -- FIXME
-    -- We also need to support the special arg 'local' for the bridge... or not.
-    parseArg (longOption "network") >>= \case
-        Testnet ->
-            runCliCommand env manager
-                (execServe @'Testnet verbosity listen backendPort dbFile)
-                (execLaunch "testnet" verbosity listen backendPort stateDir)
-        Mainnet ->
-            runCliCommand env manager
-                (execServe @'Mainnet verbosity listen backendPort dbFile)
-                (execLaunch "mainnet" verbosity listen backendPort stateDir)
+main :: forall (n :: Network). IO ()
+main = join $ execParser $ cli $ mempty
+    <> cmdLaunch
+    <> cmdServe
+    <> cmdMnemonic
+    <> cmdWallet @(HttpBridge n)
+    <> cmdTransaction @(HttpBridge n)
+    <> cmdAddress @(HttpBridge n)
+    <> cmdVersion
 
 {-------------------------------------------------------------------------------
-                               CLI definitions
--------------------------------------------------------------------------------}
-
-cliDefinition :: Docopt
-cliDefinition = makeCli
-    cliCommands
-    cliOptions
-    cliExamples
-
-cliCommands :: String
-cliCommands = [here|
-  cardano-wallet launch [--network=STRING] [(--port=INT | --random-port)] [--backend-port=INT] [--state-dir=DIR] [(--quiet | --verbose )]
-  cardano-wallet serve  [--network=STRING] [(--port=INT | --random-port)] [--backend-port=INT] [--database=FILE] [(--quiet | --verbose )]
-|]
+                            Command - 'launch'
 
-cliOptions :: String
-cliOptions = [here|
-  --backend-port <INT>      port used for communicating with the HTTP bridge [default: 8080]
-|]
+  cardano-wallet launch
+    [--network=STRING]
+    [(--port=INT | --random-port)]
+    [--node-port=INT]
+    [--state-dir=DIR]
+    [(--quiet | --verbose )]
```

Comment:
The comments are more of a "specification". I am thinking about turning this into a unit test actually because, one of the "downside" of using `optparse-applicative` (the only downside I think...) is that, it can get quite hard to figure out what your CLI actually does and allow... 
We had the case in the past where, no one would really know where options would come from and what use would they have. 

So, the idea I think is to still have a docopt-like specification for our CLI (and its various level) and maybe, have unit tests enforcing that. This way, we keep a little control on what the CLI becomes and we can ask ourselves when something changes: is that change really okay and consistent with the rest?

Said differently, the docopt-like  specification gives us a high-level overview of the CLI which we otherwise don't get that easily with `optparse-applicative`.

Reply:
>  I am thinking about turning this into a unit test actually because, one of the "downside" of using optparse-applicative (the only downside I think...) is that, it can get quite hard to figure out what your CLI actually does and allow...

I agree here, it would be a good idea to encode this specification as a unit test, rather than writing the specification in comments (which can neither be checked by the compiler nor be executed by a test).